### PR TITLE
Gsoc develop merge

### PR DESCRIPTION
**Pull request by aravindm711**
I'm still working on refactoring the tests for the recent changes.

### CODE REVIEW

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
Armin Rigo explains why the tests fail for xdist here
https://stackoverflow.com/questions/25958659/py-test-with-xdist-is-not-executing-tests-parametrized-with-random-values

It would be good to find a solution that we can keep those tests for xdist too.

This needs to be done in a new PR, this one is a merge

**Review by ReimarBauer — COMMENTED** [review]
did a dir diff to origin. have not seen conflicts by that on none project related files

**Review by ReimarBauer — COMMENTED** [review]
The flake8 failure may get solved, may be occured by a merge conflict?
https://github.com/Open-MSS/MSS/blob/GSOC2021-AravindMurali/mslib/msui/_tests/test_mscolab.py

File: mslib/msui/_tests/test_mscolab.py [hunk 1]
```diff
@@ -344,6 +401,21 @@ def test_close_help_dialog(self):
...
+    def test_create_dir_exceptions(self, mockexit, mockbox):
+        with mock.patch("fs.open_fs", new=ExceptionMock(fs.errors.CreateFailed).raise_exc):
+            self.window.data_dir = "://"
+            self.window.create_dir()
+            assert mockbox.critical.call_count == 1
+            assert mockexit.call_count == 1
+
+        with mock.patch("fs.open_fs", new=ExceptionMock(fs.opener.errors.UnsupportedProtocol).raise_exc):
+            self.window.data_dir = "://"
+            self.window.create_dir()
```
Comment (ReimarBauer) [inline mslib/msui/_tests/test_mscolab.py]: self.main_window ?

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
see comments

I guess some problems are given because of solved merge conflicts.

Fix those please

File: mslib/msui/_tests/test_mscolab.py (hunk 1 above)
Comment (aravindm711) [reply mslib/msui/_tests/test_mscolab.py]: I have fixed this issue. I'll push the commit by today evening.

**Comment by aravindm711** [comment]
I have fixed both flake8 failures and create_dir test failure, working on only other failing test, test_import_file. Will push the commit by today evening. @ReimarBauer

**Comment by aravindm711** [comment]
Work asynchronously import now supports all plugins.

**Pull request merged**